### DISCUSSION TITLE
Question about Lazygit and Commitizen

### DISCUSSION THREAD

**Discussion by zer0cha0s** [Matt-FTW/dotfiles#11] [Q&A] [answered]
Hey, just wanted to start off by saying the work you have put into your setup is next level, i fully switched to Linux last may and have worked my way from EndevourOS to Arch and the last couple of months has been me removing my reliance on a mouse for day to day usage, your dotfiles have been a great source of learning for me.

I wondered if your able to shed some light on an issue I'm having?
When using Lazygit via the dots scratchpad the Shift+c commit command returns `git: 'cz' is not a git command`. but it if i run `yadm enter lazygit` manually then it works fine (it also works for normal repo's as well). 

Did you encounter this problem when setting your system up or would you have any idea what would be causing it? I've been going round in circles for several hours today trying to work out whats wrong with it and i'm having zero luck.

**Comment by Matt-FTW** [comment]
Hi! :wave:
Im glad my work has been a great learning tool :)

About the issue, is weird because the scratchpad action is the same as executing the command in a normal terminal (With all of the shell variables loaded afaik). Maybe it could be an issue with how cz-git is installed in your system (the shell is not picking up the path for the binary?) 

You could open the dots scratchpad and then, open a new tab and find if cz is executable.
Where do you have cz-git installed? Is it on the XDG_DATA_HOME/npm/bin or is it on the /usr/bin/node_modules folder?

Tell me if you find anything

**Reply by zer0cha0s** [reply]
Thanks for taking the time to reply. 

- Commitizen is installed globally
- I've tried cz-git both locally and globally. 
- `git cz ` does work if I'm in a git repo as does `yadm cz`.
- `git cz` does not work in a yadm initialized folder and lazygit's shift+C is calling `git cz` 


Screenshot of the contents of my .local/share/npm/bin directory.
![image](https://github.com/Matt-FTW/dotfiles/assets/21130965/1cce5f16-4d4f-4d72-b1cc-d7edbdf95adb)

Error im getting :
![image](https://github.com/Matt-FTW/dotfiles/assets/21130965/a9a63bec-65c1-4dd8-9337-ece171acec92)

Maybe i messed something up or i'm not understanding how to use yadm correctly.

**Reply by Matt-FTW** [reply]
Thats weird, does lazygit calling cz work on a normal repo?

**Reply by zer0cha0s** [reply]
Yea that works fine.
Not to worry though i'm sure ill work out whats causing it soon enough. 
Just thought ask check in case it was something you had seen while tinkering with your set up.

**Reply by Matt-FTW** [reply]
Ok then I'll close the discussion. 
Tell me if you find what was the problem

**Reply by zer0cha0s** [reply, marked as answer]
@Matt-FTW So the issue was my default shell was still set to zsh. 
Kitty was loading the fish shell profile but Lazygit custom commands was prefixing the commands with bash so it was picking up zsh.
Lazygit is now working as i wanted now.

**Reply by Matt-FTW** [reply]
Oooooh, glad you found the issue!
